### ISSUE TITLE
Enhance parseq-rest-client to support cache of rest.li requests to reduce fanout calls.

### ISSUE THREAD

**Issue by mchen07** [linkedin/parseq#193]
Sometimes application make duplicate downstream calls when serving a request. One way is to make the Task sharable and pass it around, but it’s sometimes hard to refactor the code while maintaining code readability. It would be great that parseq restli client can provide some cache capability to avoid duplicate calls.

**Comment by hiteshsharma** [comment]
This doesn't have to be a cache but could be a client supplied unique identifier, for each task, which Parseq can use before executing the plan and can de-dupe tasks. Clients should be responsible for providing unique identifier for non-identical tasks and unique ones for identical tasks. This will provide clients more control on the behaviour. Adding cache would also require client to have control of the cache parameters as it can affect the machine it's running on.

**Comment by mchen07** [comment]
@hiteshsharma I don't quite get your comments. In this proposed approach, ParSeq is maintaining the cache and ask for clients to provide cache key (as you indicated above unique identifier) K.

**Comment by hiteshsharma** [comment]
@mchen07 later I realized a cache would anyways be required. Pardon my limited knowledge of Parseq. I'm really looking forward to this change. If this is being picked up I can also provide some help with some sub-tasks if needed.

**Comment by aliciatang** [comment]
Does it already exist in parseq-rest client already?

**Comment by mchen07** [comment]
No, I don't think ParSeq-rest-client has cache built in. @karthikbalasub @junchuanwang Do we have any plans to work on this in the near future?

**Comment by junchuanwang** [comment]
> No, I don't think ParSeq-rest-client has cache built in. @karthikbalasub @junchuanwang Do we have any plans to work on this in the near future?

@mchen07  We just discussed yesterday and from my understanding we are not going to work on this at least this quarter.
